plutus-tx/src/Language/PlutusTx/Lift.hs

```diff
@@ -44,26 +45,37 @@ type Throwable uni = (PLC.GShow uni, PLC.Closed uni, uni `PLC.Everywhere` Pretty
 
 -- | Get a Plutus Core term corresponding to the given value.
 safeLift
-    :: (Lift.Lift uni a, AsError e uni (Provenance ()), MonadError e m, MonadQuote m)
+    :: (Lift.Lift uni a
+       , PIR.AsTypeError e uni (Provenance ()), PLC.GShow uni, PLC.GEq uni
+       , PLC.DefaultUni PLC.<: uni
+       , AsError e uni (Provenance ()), MonadError e m, MonadQuote m)
     => a -> m (PLC.Term TyName Name uni ())
 safeLift x = do
     lifted <- liftQuote $ runDefT () $ Lift.lift x
-    compiled <- flip runReaderT defaultCompilationCtx $ compileTerm lifted
+    -- FIXME: we need to pass the real dynamic builtin tcconfig map in compileterm
```

Comment:
I think this may actually be fine: `Lift` currently never uses builtin functions, since it only ever constructs "data". It does use builtin types, but they come from the universe so that's fine.